src/window.c

```diff
@@ -1900,42 +1918,59 @@ win_rotate(int upwards, int count)
 }
 
 /*
- * Move the current window to the very top/bottom/left/right of the screen.
+ * Move "wp" into a new split in a given direction, possibly relative to the
+ * current window.
+ * "wp" must be valid in the current tabpage.
+ * Returns FAIL for failure, OK otherwise.
  */
-    static void
-win_totop(int size, int flags)
+    int
+win_splitmove(win_T *wp, int size, int flags)
 {
     int		dir;
-    int		height = curwin->w_height;
+    int		height = wp->w_height;
+    frame_T	*unflat_altfr;
 
     if (ONE_WINDOW)
-    {
-	beep_flush();
-	return;
-    }
-    if (check_split_disallowed() == FAIL)
-	return;
+	return OK;	// nothing to do
+    if (check_split_disallowed(wp) == FAIL)
```

Comment:
This is consistent with how `win_totop` worked, but maybe it's not necessary to check the `wp->w_buffer->b_locked_split` part before split-moving, as it won't produce many views into the same buffer (which I think was the motivation for the flag in [v8.2.2476](https://github.com/vim/vim/commit/983d83ff1cd796ff321074335fa53fbe7ac45a46)). I haven't checked this, though.